src/Conversion/KrnlToLLVM/KrnlToLLVMHelper.cpp

```diff
@@ -342,5 +343,47 @@ bool isZOS(ModuleOp module) {
   return zOS;
 }
 
+void equalOrFailed(ModuleOp &module, OpBuilder &rewriter, Location loc,
```

Comment:
Would these calls also be potentially useful if we wanted to fail the inference, say after a NNPA severe failure call?
If so, we may want to see if we would benefit from introducing something at a higher level dialects (say Krnl or SCF) that would then feed into this one.

Reply:
Totally yes. It will be used anywhere in the main inference when we want to fail the inference. We can introduce a krnl op for this.